crates/ra_assists/src/merge_match_arms.rs

```diff
@@ -0,0 +1,189 @@
+use hir::db::HirDatabase;
+use ra_syntax::{
+    ast::{AstNode, MatchArm},
+    algo::non_trivia_sibling, Direction, T
+};
+use crate::{Assist, AssistCtx, AssistId};
+
+pub(crate) fn merge_match_arms(mut ctx: AssistCtx<impl HirDatabase>) -> Option<Assist> {
+    let comma = ctx.token_at_offset().find(|leaf| leaf.kind() == T![,])?;
```

Comment:
I think we should just look at `MatchArm`  here, comma is not required if the body of the arm is in braces

```
match x {
    1 => { println!("hello"); }
    2 => { println!("hello"); }
}
```